jdbc/src/main/java/org/jclouds/jdbc/repository/ContainerRepository.java

```diff
@@ -19,32 +19,40 @@
 import com.google.inject.Inject;
 import com.google.inject.Provider;
 import com.google.inject.Singleton;
-import org.jclouds.jdbc.entity.Container;
+import org.jclouds.jdbc.entity.ContainerEntity;
 
 import javax.persistence.EntityManager;
+import javax.persistence.PersistenceException;
 import java.util.List;
 
 @Singleton
-public class ContainerRepository extends GenericRepository<Container, Long> {
+public class ContainerRepository extends GenericRepository<ContainerEntity, Long> {
 
    @Inject
    private ContainerRepository(Provider<EntityManager> entityManager) {
       super(entityManager);
    }
 
-   public Container findContainerByName(String name) {
-      return entityManager.get().createQuery("SELECT c FROM " + entityClass.getName() + " c WHERE c.name = :name", entityClass)
-            .setParameter("name", name)
-            .getSingleResult();
+   public ContainerEntity findContainerByName(String name) {
+      try {
+         return entityManager.get().createQuery("SELECT c FROM " + entityClass.getName() + " c WHERE c.name = :name", entityClass)
+               .setParameter("name", name)
+               .getSingleResult();
+      } catch (PersistenceException e) {
+         return null;
```

Comment:
You'd better just catch the `NoResultException` to return null and propagate the exception otherwise.